src/Functional.CQS.AOP.IoC.SimpleInjector/DecoratorRegistrationGateways/ConventionBasedDecoratorRegistrationGateway.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Reflection;
+using SimpleInjector;
+using Container = SimpleInjector.Container;
+
+namespace Functional.CQS.AOP.IoC.SimpleInjector.DecoratorRegistrationGateways
+{
+	/// <summary>
+	/// Gateway for applying decorators to IQ.Vanilla.CQS handler implementations that have been registered via convention.
+	/// </summary>
+	public sealed class ConventionBasedDecoratorRegistrationGateway
+	{
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ConventionBasedDecoratorRegistrationGateway"/> class.
+		/// </summary>
+		/// <param name="container">The container containing prior registration of IQ.Vanilla.CQS handler implementations.</param>
```

Comment:
`IQ.Vanilla.CQS` should probably be changed to `Functional.CQS`.

Reply:
Same comment applies to the description of the `lifestyle` parameter on this method.